### DISCUSSION TITLE
How to parse json?

### DISCUSSION THREAD

**Discussion by DanilKrivonos** [antismash/antismash#533] [Q&A]
Hello, thank you for your wonderful tool! I would like to ask you about a better way for parsing antismash json format. It looks like some output jsones have different data fields (for example, Protein ID or Gene), this makes iterative data parsing difficult. In addition, he also wonders what this may be related to, why somewhere certain fields have values, but somewhere they don’t, while it even have not such a key in the final file.
Thanks,
Danil.

**Comment by SJShaw** [comment]
I am assuming you're looking at the record details, which are generally a copy of the input(s).

The `protein_id`, `gene` and `locus_tag` fields of a CDS are populated directly from the input. It's quite common for genbanks to not provide all three, and sometimes not using them consistently.

Within antiSMASH itself, we use the first value of `locus_tag`, `gene`, and `protein_id` which is set, and it must be unique (very occasionally for splice variants, we need to do some extra modification of the name to make it unique).

If my assumption was wrong, please provide have an example section of JSON.
